File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportJava/Networking/src/main/java/transport/networking/protocol/Request.java
package transport.networking.protocol;

import java.io.Serializable;

public class Request implements Serializable {
    private RequestType type;
    private Object data;

    public Request(RequestType type, Object data) {
        this.type = type;
        this.data = data;
    }

    public RequestType getType() {
        return type;
    }

    public Object getData() {
        return data;
    }
}

File: Semestrul 4/Medii de proiectare si programare/Laboratoare/FirmaTransportC#/Network/protocol/Request.cs
using FirmaTransportC_.Networking.dto;

namespace FirmaTransportC_.Networking.protocol
{
    [Serializable]
    public class Request
    {
        public RequestType Type { get; set; }
        public UserDTO? User { get; set; }
        public TripDTO? Trip { get; set; }
        public SearchDTO? SearchParams { get; set; }
        public ReservationDTO? Reservation { get; set; }

        public override string ToString() => $"Request{{Type={Type}, User={User}}}";
    }
}
